provisioning/service/src/Authentication/ServiceConnectionString.cs

```diff
@@ -1,156 +1,235 @@
+
+        private string _hostName;
+        private string _serviceName;
+
+        private ServiceConnectionString()
+        {
+        }
 
         /// <summary>
-        /// CONSTRUCOR
+        /// Factory for new Connection String object.
         /// </summary>
-        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
-        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
-        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        /// <remarks>
+        /// The connection string contains a set of information that uniquely identify an IoT Service.
+        /// 
+        /// A valid connection string shall be in the following format:
+        /// <code>
+        /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+        /// </code>
+        /// 
+        /// This object parse the connection string providing the artifacts to the <see cref="ServiceConnectionString"/> object. 
+        /// </remarks>
+        /// <param name="serviceConnectionString">the <code>string</code> with the connection string information.</param>
+        /// <returns>A <code>ServiceConnectionStringParser</code> object with the parsed connection string.</returns>
+        public static ServiceConnectionString Create(string serviceConnectionString)
         {
-            if (parser == null)
+            if (string.IsNullOrWhiteSpace(serviceConnectionString))
             {
-                throw new ArgumentNullException(nameof(parser));
+                throw new ArgumentNullException(nameof(serviceConnectionString));
             }
 
-            HostName = parser.HostName;
-            SharedAccessKeyName = parser.SharedAccessKeyName;
-            SharedAccessKey = parser.SharedAccessKey;
-            SharedAccessSignature = parser.SharedAccessSignature;
-            ServiceName = parser.ServiceName;
-            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
-        }
+            var ServiceConnectionStringParser = new ServiceConnectionString();        
+            ServiceConnectionStringParser.Parse(serviceConnectionString);
 
-        /// <summary>
-        /// The Provisioning Service Name
-        /// </summary>
-        public string ServiceName
-        {
-            get;
-            private set;
+            return ServiceConnectionStringParser;
         }
 
         /// <summary>
         /// The Provisioning Service Client Hostname
         /// </summary>
         public string HostName
         {
-            get;
-            private set;
+            get { return _hostName; }
+            set { SetHostName(value); }
         }
 
         /// <summary>
-        /// The Provisioning Service Client Https Endpoint
+        /// The Provisioning Service Access Policy Name
```

Comment:
Why is it policy name (like in portal) but the API is "Key Name"? This is very confusing.